### PR TITLE
add validator TRX addresses (TRC20)

### PR DESCRIPTION
**Pull request by msamsami**
This is a validator for TRC20 Tron (TRX) addresses.

### CODE REVIEW

**Comment by joe733** [comment]
Hi @msamsami thanks for the PR.

Can you move both #276 & #277 to a folder `crypto_addresses` under `validators`, like so?

```c
.
|---- validators
    |---- crypto_addresses
        |---- eth_address.py
        |---- trx_address.py
```

And do the same with their corresponding tests?

**Comment by msamsami** [comment]
Hi @joe733 

Sure. I'll move `btc_address.py` and its test to the new directories as well.

**Comment by msamsami** [comment]
> Hi @msamsami thanks for the PR.
> 
> Can you move both #276 & #277 to a folder `crypto_addresses` under `validators`, like so?
> 
> ```c
> .
> |---- validators
>     |---- crypto_addresses
>         |---- eth_address.py
>         |---- trx_address.py
> ```
> 
> And do the same with their corresponding tests?

Hi @joe733 

It's done. Please merge #276 and then #277 to avoid any conflicts. The final structure would be like this:

```c
.
|---- tests
    |---- crypto_addresses
        |---- __init__.py
        |---- test_btc_address.py
        |---- test_eth_address.py
        |---- test_trx_address.py

|---- validators
    |---- crypto_addresses
        |---- __init__.py
        |---- btc_address.py
        |---- eth_address.py
        |---- trx_address.py

```

**Review by joe733 — CHANGES_REQUESTED** [review]
Shouldn't PR #277 be tracking TRX addresses?

File: poetry.lock [hunk 1]
```diff
@@ -1099,6 +1099,37 @@ nodeenv = ">=1.6.0"
 all = ["twine (>=3.4.1)"]
 dev = ["twine (>=3.4.1)"]
 
+[[package]]
+name = "pysha3"
```
[the remark this answers is not in the document]
Comment (msamsami) [reply poetry.lock:1103]: I will restructure this PR after #276 is accepted.

**Comment by joe733** [comment]
@msamsami please sync `add-trx-validator` branch from upstream.

**Comment by yozachar** [comment]
Ref: https://github.com/python-validators/validators/pull/278#issuecomment-2036013050

**Pull request closed without merging**